libraries/Microsoft.Bot.Builder.Dialogs/Choices/ChoiceFactory.cs

```diff
@@ -4,42 +4,50 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Text;
 using Microsoft.Bot.Builder.Core.Extensions;
 using Microsoft.Bot.Schema;
 
 namespace Microsoft.Bot.Builder.Dialogs.Choices
 {
     public class ChoiceFactory
     {
-        public static IMessageActivity ForChannel(ITurnContext context, List<Choice> choices, string text = null, string speak = null, ChoiceFactoryOptions options = null)
+        public static IMessageActivity ForChannel(ITurnContext context, IEnumerable<Choice> choices, string text = null, string speak = null, ChoiceFactoryOptions options = null)
         {
             return ForChannel(Channel.GetChannelId(context), choices, text, speak, options);
         }
 
-        public static IMessageActivity ForChannel(ITurnContext context, List<string> choices, string text = null, string speak = null, ChoiceFactoryOptions options = null)
+        public static IMessageActivity ForChannel(ITurnContext context, IEnumerable<string> choices, string text = null, string speak = null, ChoiceFactoryOptions options = null)
         {
             return ForChannel(Channel.GetChannelId(context), ToChoices(choices), text, speak, options);
         }
 
-        public static IMessageActivity ForChannel(string channelId, List<string> choices, string text = null, string speak = null, ChoiceFactoryOptions options = null)
+        public static IMessageActivity ForChannel(string channelId, IEnumerable<string> choices, string text = null, string speak = null, ChoiceFactoryOptions options = null)
         {
             return ForChannel(channelId, ToChoices(choices), text, speak, options);
         }
 
-        public static IMessageActivity ForChannel(string channelId, List<Choice> list, string text = null, string speak = null, ChoiceFactoryOptions options = null)
+        public static IMessageActivity ForChannel(string channelId, IEnumerable<Choice> choices, string text = null, string speak = null, ChoiceFactoryOptions options = null)
         {
-            list = list ?? new List<Choice>();
+            if (choices is null)
+            {
+                throw new ArgumentNullException("The collection of choices can not be null.", nameof(choices));
+            }
+
+            var list = new List<Choice>();
 
             // Find maximum title length
             var maxTitleLength = 0;
-            foreach (var choice in list)
+            foreach (var choice in choices)
             {
-                var l = choice.Action != null && string.IsNullOrEmpty(choice.Action.Title) ? choice.Action.Title.Length : choice.Value.Length;
-                if (l > maxTitleLength)
+                int len = GetTitle(choice).Length;
```

Comment:
These changes seem to be a behaviorial. I don't see a revised test for this. 

I would suggest backing this change out of the PR, making this PR just abou the the IList -> IEnumerable change, and doing the behavioral changes in a different PR.